File: CS425/CS425SpectralAnalysisScript.m
close all;
playSoundFlag=0;
Fs=11025; #set the sampling frequency
fhz=1110; #set the frequency in Hertz
tdur=1;  #set the time duration in seconds
Amplitude=1; #amplitude of the wave
phase=0; #Phase shift in radians;

%generate a single sinewave
[output1s,t]=Gen_sine(Amplitude,fhz,Fs,tdur,phase);
figure(1)
plot(t(1:200),output1s(1:200)), axis tight
xlabel('Time (Sec)'), ylabel('Amplitude'),title('Single Sinewave')
if (playSoundFlag==1)
soundsc(output1s,Fs)
endif

FFTlength=2048;
FxLen1=fft(output1s,FFTlength);
Magnitude=abs(FxLen1);
Phase=angle(FxLen1);
faxis=0:Fs./FFTlength:Fs-Fs./FFTlength;
figure(2),
subplot(2,1,1),plot(faxis,Magnitude), axis tight
xlabel('Frequency (Hertz)'), ylabel('Magnitude'),title('Magnitude of Frequency Spectrum of Sinewave with FFT length=2048 and Sampling Frequency=11025')
subplot(2,1,2),plot(faxis,Phase), axis tight
xlabel('Frequency (Hertz)'), ylabel('Phase'),title('Phase of Frequency Spectrum of Sinewave with FFT length=2048 and Sampling Frequency=11025')
FreqResolution=num2str(Fs/FFTlength);
message=cstrcat("Frequency Resolution of FFT length=2048 for Fs=11025 is ", FreqResolution,"Hz\n");
printf(message)

FFTlength=8192;
FxLen2=fft(output1s,FFTlength);
Magnitude=abs(FxLen2);
Phase=angle(FxLen2);
faxis=0:Fs./FFTlength:Fs-Fs./FFTlength;
figure(3),
subplot(2,1,1),plot(faxis,Magnitude), axis tight
xlabel('Frequency (Hertz)'), ylabel('Magnitude'),title('Magnitude of Frequency Spectrum of Sinewave with FFT length=2048 and Sampling Frequency=11025')
subplot(2,1,2),plot(faxis,Phase), axis tight
xlabel('Frequency (Hertz)'), ylabel('Phase'),title('Phase of Frequency Spectrum of Sinewave with FFT length=2048 and Sampling Frequency=11025')
FreqResolution=num2str(Fs/FFTlength);
message=cstrcat("Frequency Resolution of FFT length=2048 for Fs=8192 is ", FreqResolution,"Hz\n");
printf(message)

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%% demonstrate the meaning of resolution %%%%%%%%%%%%%%%%%%%%%%%%%
Fs=11025;
fhz1=1110; #set the frequency in Hertz
fhz2=1114; #set the frequency in Hertz
tdur=1;  #set the time duration in seconds
Amplitude=0.5; #amplitude of the wave
phase=0; #Phase shift in radians;

%generate a single sinewave
[output1s,t]=Gen_sine(Amplitude,fhz1,Fs,tdur,phase);
[output2s,t]=Gen_sine(Amplitude,fhz2,Fs,tdur,phase);
outputTotal=output1s+output2s;
figure(4)
plot(t(1:200),outputTotal(1:200)), axis tight
xlabel('Time (Sec)'), ylabel('Amplitude'),title('Two Sinewaves only 2Hz apart')
if (playSoundFlag==1)
soundsc(outputTotal,Fs)
endif

FFTlength=2048;
FxLen1=fft(outputTotal,FFTlength);
Magnitude=abs(FxLen1);
faxis=0:Fs./FFTlength:Fs-Fs./FFTlength;
figure(5),
subplot(2,1,1),
plot(faxis(1:FFTlength/2),Magnitude(1:FFTlength/2)), axis([1090 1130 0 600])
xlabel('Frequency (Hertz)'), ylabel('Magnitude'),title('Zoomed Magnitude of Frequency Spectrum of two Sinewaves only 4Hz apart with FFT Resolution of 5.4Hz')
FreqResolution=num2str(Fs/FFTlength);
message=cstrcat("When the Frequency Resolution is ", FreqResolution,"Hz only one distinct peak can be seen in the magnitude spectrum\n");
printf(message)

FFTlength=8192;
FxLen2=fft(outputTotal,FFTlength);
Magnitude=abs(FxLen2);
faxis=0:Fs./FFTlength:Fs-Fs./FFTlength;
subplot(2,1,2),
%figure(6)
plot(faxis(1:FFTlength/2),Magnitude(1:FFTlength/2)), axis([1090 1130 0 2200])
xlabel('Frequency (Hertz)'), ylabel('Magnitude'),title('Zoomed Magnitude of Frequency Spectrum of two Sinewaves only 2Hz apart with FFT Resolution of 1.3Hz')
FreqResolution=num2str(Fs/FFTlength);
message=cstrcat("When the Frequency Resolution is ", FreqResolution,"Hz two distinct peaks can be seen in the magnitude spectrum\n");
printf(message)

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%% dB Analysis%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Fs=11025; #set the sampling frequency
fhz=1110; #set the frequency in Hertz
tdur=1;  #set the time duration in seconds
Amplitude=1; #amplitude of the wave
phase=0; #Phase shift in radians;

%generate a single sinewave
[output1s,t]=Gen_sine(Amplitude,fhz,Fs,tdur,phase);

FFTlength=2048;
FxLen1=fft(output1s,FFTlength);
Magnitude=abs(FxLen1);
MagnitudedB=20.*log10(Magnitude);

faxis=0:Fs./FFTlength:Fs-Fs./FFTlength;
figure(6),
subplot(2,1,1),plot(faxis(1:FFTlength/2),Magnitude(1:FFTlength/2)), axis tight
xlabel('Frequency (Hertz)'), ylabel('Magnitude'),title('Magnitude of Frequency Spectrum of Sinewave with FFT length=2048 and Sampling Frequency=11025')
subplot(2,1,2),plot(faxis(1:FFTlength/2),MagnitudedB(1:FFTlength/2)), axis tight
xlabel('Frequency (Hertz)'), ylabel('Magnitude (dB)'),title('Magnitude in dB of Frequency Spectrum of Sinewave with FFT length=2048 and Sampling Frequency=11025')

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%% windowing %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
pkg load signal;
Fs=11025; #set the sampling frequency
fhz=1110; #set the frequency in Hertz
tdur=1;  #set the time duration in seconds
Amplitude=1; #amplitude of the wave
phase=0; #Phase shift in radians;

%generate a single sinewave
[output1s,t]=Gen_sine(Amplitude,fhz,Fs,tdur,phase);

FFTlength=2048;
sigLen=2000;
win=rectwin(sigLen)';
winSig=output1s(1:sigLen).*win;
FxLen1win=fft(winSig,FFTlength);
Magnitude=abs(FxLen1win);
MagnitudedB=20.*log10(Magnitude);

faxis=0:Fs./FFTlength:Fs-Fs./FFTlength;
figure(7),
subplot(2,1,1),
plot(faxis(1:FFTlength/2),MagnitudedB(1:FFTlength/2)), axis tight
xlabel('Frequency (Hertz)'), ylabel('Magnitude (dB)'),title('Magnitude of Frequency Spectrum of Sinewave with FFT length=2048, Rectangular window')

figure(8),
subplot(2,1,1),plot(faxis(1:FFTlength/2),MagnitudedB(1:FFTlength/2)), axis([1000 2000 -10 60])
xlabel('Frequency (Hertz)'), ylabel('Magnitude (dB)'),title('Zoomed Magnitude of Frequency Spectrum of Sinewave with FFT length=2048, Rectangular window')
message=cstrcat("Using a rectangular window the peak is narrower but the spectral sidelobes are higher\n");
printf(message)

sigLen=2000;
win=hanning(sigLen)';
winSig=output1s(1:sigLen).*win;
FxLen1win=fft(winSig,FFTlength);
Magnitude=abs(FxLen1win);
MagnitudedB=20.*log10(Magnitude);
figure(7)
subplot(2,1,2),plot(faxis(1:FFTlength/2),MagnitudedB(1:FFTlength/2)), axis tight
xlabel('Frequency (Hertz)'), ylabel('Magnitude (dB)'),title('Magnitude in dB of Frequency Spectrum of Sinewave with FFT length=2048, Hanning window')

figure(8),
subplot(2,1,2),plot(faxis(1:FFTlength/2),MagnitudedB(1:FFTlength/2)), axis([1000 2000 -10 60])
xlabel('Frequency (Hertz)'), ylabel('Magnitude (dB)'),title('Zoomed Magnitude of Frequency Spectrum of Sinewave with FFT length=2048, Hannign window')
message=cstrcat("Using a Hanning window the peak is broader but the spectral sidelobes are lower\n");
printf(message)

figure(9)
subplot(2,1,1), plot([zeros(128,1);rectwin(2048);zeros(128,1)] ), axis([0 2304 0 1.5]);
xlabel('Sample number'),title('Rectangular Window')
subplot(2,1,2), plot([zeros(128,1); hanning(2048);zeros(128,1)]), axis([0 2304 0 1.5]);
xlabel('SAmple number'),title('Hanning Window')


%%% windowing of a wave with multiple sines
Fs=11025; %set the sampling frequency
fhzVect=[110 220 330 440 550 660]; %set a vector of frequencies in Hertz
tdur=1; %set the time duration in seconds
AmplitudeVect=[1 1/2 1/3 1/4 1/5 1/6]; %set a vector of amplitudes
phaseShiftVect=[0 0 0 0 0 0]; %Set a vector of Phase shifts in radians;
[output2,t]=Gen_MultiSine(AmplitudeVect,fhzVect,Fs,tdur,phaseShiftVect);
FFTlength=8192;
sigLen=4000;
win=rectwin(sigLen)';
winSig=output2(1:sigLen).*win;
FxLen1win=fft(winSig,FFTlength);
Magnitude=abs(FxLen1win);
MagnitudedB=20.*log10(Magnitude);

faxis=0:Fs./FFTlength:Fs-Fs./FFTlength;
figure(10),
subplot(2,1,1),plot(faxis(1:FFTlength/2),MagnitudedB(1:FFTlength/2)), axis([0 1500 -20 70]) 
xlabel('Frequency (Hertz)'), ylabel('Magnitude (dB)'),title('Magnitude of Frequency Spectrum of MultiSine with FFT length=8192, Rectangular window')
message=cstrcat("Using a rectangular window the spectral sidelobes/noise floor is much higher\n");
figure(11),
subplot(2,1,1),plot(faxis(1:FFTlength/2),MagnitudedB(1:FFTlength/2)), axis([0 800 20 70])
xlabel('Frequency (Hertz)'), ylabel('Magnitude (dB)'),title('Magnitude of Frequency Spectrum of MultiSine with FFT length=8192, Rectangular window')
sigLen=4000;
win=hanning(sigLen)';
winSig=output2(1:sigLen).*win;
FxLen1win=fft(winSig,FFTlength);
Magnitude=abs(FxLen1win);
MagnitudedB=20.*log10(Magnitude);
figure(10)
subplot(2,1,2),plot(faxis(1:FFTlength/2),MagnitudedB(1:FFTlength/2)), axis([0 1500 -20 70]) 
xlabel('Frequency (Hertz)'), ylabel('Magnitude (dB)'),title('Magnitude in dB of Frequency Spectrum of MultiSine with FFT length=8192, Hanning window')
figure(11)
subplot(2,1,2),plot(faxis(1:FFTlength/2),MagnitudedB(1:FFTlength/2)), axis([0 800 20 70]) 
xlabel('Frequency (Hertz)'), ylabel('Magnitude (dB)'),title('Magnitude in dB of Frequency Spectrum of MultiSine with FFT length=8192, Hanning window')

message=cstrcat("Using a Hanning window the spectral sidelobes/noise floor are lower\n");

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%5 Spectral Analysis of Noise %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%Generate Noise
Fs=11025;
wn=randn(1,11025);
FFTlength=4096;
FxLenWn=fft(wn,FFTlength);
MagnitudeWn=abs(FxLenWn);
MagnitudeWndB=20.*log10(abs(FxLenWn));
faxis=0:Fs./FFTlength:Fs-Fs./FFTlength;
figure(12),
plot(faxis(1:FFTlength/2),MagnitudeWndB(1:FFTlength/2)), axis tight
xlabel('Frequency (Hertz)'), ylabel('Magnitude (dB)'),title('Magnitude of Frequency Spectrum of White Noise FFT length=4096')
message=cstrcat("Note of the flat spectrum of White Noise\n");

File: CS425/Java files for Spectral Analysis-20200529/CS425SpectralAnalysisScript.java
package Frequency;

import java.util.Random;
import java.util.Scanner;

import signal.plotXYline;
import signal.signalGen;
import signal.stdAudio;

public class CS425SpectralAnalysisScript {
public static int playSoundFlag=1;

	public static void main(String args[]) {
	Scanner scan = new Scanner(System.in);
    System.out.print("Enter the plot number: ");
    int plotNumber = scan.nextInt();
    // Closing Scanner after the use
    scan.close(); 
    
	if (plotNumber==1) { //Sinusoidal signal  frequency = 1110Hz, Fs=11025, and FFTlength=2048 
	signalGen sig_s=new signalGen();
	plotXYline makePlot_s=new plotXYline();
	FFT fourierTransform=new FFT();
	double fhz_s=1110;
	double Fs_s=11025;
	double tdur_s=1;
	double Amplitude_s=1;
	double phaseShift_s=0;
	double[] output1s=sig_s.SineGen(Amplitude_s,fhz_s,Fs_s,tdur_s,phaseShift_s);
	double[] twoPeriodsSignal=makePlot_s.plotTwoPeriodsSignal(output1s, Fs_s, fhz_s);
	double[] twoPeriodsTimeAxis=makePlot_s.plotTimeAxisTwoPeriods(output1s, Fs_s, fhz_s);
	makePlot_s.plotData(twoPeriodsTimeAxis, twoPeriodsSignal, "Single Sinewave", "Time (Sec)", "Amplitude", "Signal Plots");
	
	int FFTlength=2048;
	double[][] frequencySpectrum=new double[2][FFTlength];
	double[] Magnitude=new double[FFTlength];
	double[] Phase=new double[FFTlength];
	double[] faxis=new double[FFTlength];
	
	frequencySpectrum=fourierTransform.doFFT(output1s, FFTlength);
	boolean halfSpectrum=false;
	Magnitude=fourierTransform.SpecMagnitude(frequencySpectrum,halfSpectrum);
	Phase=fourierTransform.SpecPhase(frequencySpectrum,halfSpectrum);
	faxis=fourierTransform.makeFrequencyAxis(Fs_s,FFTlength,halfSpectrum);
	System.out.println(Magnitude.length);
	makePlot_s.plotData(faxis, Magnitude, "Magnitude of Frequency Spectrum of Sinewave with FFT length=2048 and Sampling Frequency=11025", "Frequency (Hz)", "Magnitude", "Spectral Plots");
	makePlot_s.plotData(faxis, Phase, "Phase of Frequency Spectrum of Sinewave with FFT length=2048 and Sampling Frequency=11025", "Frequency (Hz)", "Magnitude", "Spectral Plots");
	
	String FreqResolution=Double.toString(Fs_s/(double) FFTlength);
	String message="Frequency Resolution of FFT length=2048 for Fs=11025 is "+ FreqResolution+"Hz";
	System.out.println(message);

	} else if (plotNumber==2) { //Sinusoidal signal frequency = 1110Hz, Fs=11025, and FFTlength=8192
	signalGen sig_s=new signalGen();
	plotXYline makePlot_s=new plotXYline();
	FFT fourierTransform=new FFT();
	double fhz_s=1110;
	double Fs_s=11025;
	double tdur_s=1;
	double Amplitude_s=1;
	double phaseShift_s=0;
	double[] output1s=sig_s.SineGen(Amplitude_s,fhz_s,Fs_s,tdur_s,phaseShift_s);
	double[] twoPeriodsSignal=makePlot_s.plotTwoPeriodsSignal(output1s, Fs_s, fhz_s);
	double[] twoPeriodsTimeAxis=makePlot_s.plotTimeAxisTwoPeriods(output1s, Fs_s, fhz_s);
	makePlot_s.plotData(twoPeriodsTimeAxis, twoPeriodsSignal, "Single Sinewave", "Time (Sec)", "Amplitude", "Signal Plots");
	
	int FFTlength=8192;
	double[][] frequencySpectrum=new double[2][FFTlength];
	double[] Magnitude=new double[FFTlength];
	double[] Phase=new double[FFTlength];
	double[] faxis=new double[FFTlength];
	
	frequencySpectrum=fourierTransform.doFFT(output1s, FFTlength);
	boolean halfSpectrum=false;
	Magnitude=fourierTransform.SpecMagnitude(frequencySpectrum,halfSpectrum);
	Phase=fourierTransform.SpecPhase(frequencySpectrum,halfSpectrum);
	faxis=fourierTransform.makeFrequencyAxis(Fs_s,FFTlength,halfSpectrum);
	makePlot_s.plotData(faxis, Magnitude, "Magnitude of Frequency Spectrum of Sinewave with FFT length=8192 and Sampling Frequency=11025", "Frequency (Hz)", "Magnitude", "Spectral Plots");
	makePlot_s.plotData(faxis, Phase, "Phase of Frequency Spectrum of Sinewave with FFT length=8192 and Sampling Frequency=11025", "Frequency (Hz)", "Magnitude", "Spectral Plots");
	
	String FreqResolution=Double.toString(Fs_s/(double) FFTlength);
	String message="Frequency Resolution of FFT length=8192 for Fs=11025 is "+ FreqResolution+"Hz";
	System.out.println(message);
	} else if (plotNumber==3) { //Demonstration of Resolution 
		signalGen sig_s=new signalGen();
		plotXYline makePlot_s=new plotXYline();
		FFT fourierTransform=new FFT();
		double fhz1_s=1110;
		double fhz2_s=1114;
		double Fs_s=11025;
		double tdur_s=1;
		double Amplitude_s=0.5;
		double phaseShift_s=0;
		double[] output1s=sig_s.SineGen(Amplitude_s,fhz1_s,Fs_s,tdur_s,phaseShift_s);
		double[] output2s=sig_s.SineGen(Amplitude_s,fhz2_s,Fs_s,tdur_s,phaseShift_s);
		double[] outputTotal=sig_s.addArray(output1s, output2s);
		double[] twoPeriodsSignal=makePlot_s.plotTwoPeriodsSignal(outputTotal, Fs_s, fhz1_s);
		double[] twoPeriodsTimeAxis=makePlot_s.plotTimeAxisTwoPeriods(outputTotal, Fs_s, fhz1_s);
		makePlot_s.plotData(twoPeriodsTimeAxis, twoPeriodsSignal, "Two Sinewaves only 4Hz apart", "Time (Sec)", "Amplitude", "Signal Plots");
				
		int FFTlength_2048=2048;
		double[][] frequencySpectrum_2048=new double[2][FFTlength_2048];
		double[] Magnitude_2048=new double[FFTlength_2048];
		double[] faxis_2048=new double[FFTlength_2048];
		
		frequencySpectrum_2048=fourierTransform.doFFT(output1s, FFTlength_2048);
		boolean halfSpectrum=true;
		Magnitude_2048=fourierTransform.SpecMagnitude(frequencySpectrum_2048,halfSpectrum);
		faxis_2048=fourierTransform.makeFrequencyAxis(Fs_s,FFTlength_2048,halfSpectrum);
		makePlot_s.plotData(faxis_2048, Magnitude_2048, "Magnitude of Frequency Spectrum of Two Sinewaves with FFT length=2048 and Sampling Frequency=11025", "Frequency (Hz)", "Magnitude", "Spectral Plots");
				
		String FreqResolution_2048=Double.toString(Fs_s/(double) FFTlength_2048);
		String message1="Frequency Resolution of FFT length=2048 for Fs=11025 is "+ FreqResolution_2048+"Hz";
		System.out.println(message1);
		
		int FFTlength_8192=8192;
		double[][] frequencySpectrum_8192=new double[2][FFTlength_8192];
		double[] Magnitude_8192=new double[FFTlength_8192];
		double[] faxis_8192=new double[FFTlength_8192];
		
		frequencySpectrum_8192=fourierTransform.doFFT(outputTotal, FFTlength_8192);
		Magnitude_8192=fourierTransform.SpecMagnitude(frequencySpectrum_8192,halfSpectrum);
		faxis_8192=fourierTransform.makeFrequencyAxis(Fs_s,FFTlength_8192,halfSpectrum);
		makePlot_s.plotData(faxis_8192, Magnitude_8192, "Magnitude of Frequency Spectrum of Two Sinewaves with FFT length=8192 and Sampling Frequency=11025", "Frequency (Hz)", "Magnitude", "Spectral Plots");
			
		String FreqResolution_8192=Double.toString(Fs_s/(double) FFTlength_8192);
		String message2="Frequency Resolution of FFT length=8192 for Fs=11025 is "+ FreqResolution_8192+"Hz";
		System.out.println(message2);
		} else if (plotNumber==4) { //dB magnitude analysis
			signalGen sig_s=new signalGen();
			plotXYline makePlot_s=new plotXYline();
			plotXYline makePlot_sdB=new plotXYline();
			FFT fourierTransform=new FFT();
			double fhz_s=1110;
			double Fs_s=11025;
			double tdur_s=1;
			double Amplitude_s=0.5;
			double phaseShift_s=0;
			double[] output1s=sig_s.SineGen(Amplitude_s,fhz_s,Fs_s,tdur_s,phaseShift_s);
					
			int FFTlength=2048;
			double[][] frequencySpectrum=new double[2][FFTlength];
			double[] Magnitude_Lin=new double[FFTlength];
			double[] Magnitude_dB=new double[FFTlength];
			double[] faxis=new double[FFTlength];
			
			frequencySpectrum=fourierTransform.doFFT(output1s, FFTlength);
			boolean halfSpectrum=true;
			Magnitude_Lin=fourierTransform.SpecMagnitude(frequencySpectrum,halfSpectrum);
			Magnitude_dB=fourierTransform.convert2dB(Magnitude_Lin);
			faxis	=fourierTransform.makeFrequencyAxis(Fs_s,FFTlength,halfSpectrum);
			makePlot_s.plotData(faxis, Magnitude_Lin, "Magnitude of Frequency Spectrum of Sinewave with FFT length=2048 and Sampling Frequency=11025", "Frequency (Hz)", "Magnitude", "Spectral Plots");
			
			
			makePlot_sdB.plotData(faxis, Magnitude_dB, "Magnitude in dB of Frequency Spectrum of Sinewave with FFT length=2048 and Sampling Frequency=11025", "Frequency (Hz)", "Magnitude", "Spectral Plots");
			} else if (plotNumber==5) { //Windowing
				signalGen sig_s=new signalGen();
				plotXYline makePlot_s=new plotXYline();
				FFT fourierTransform=new FFT();
				double fhz_s=1110;
				double Fs_s=11025;
				double tdur_s=1;
				double Amplitude_s=0.5;
				double phaseShift_s=0;
				double[] output1s=sig_s.SineGen(Amplitude_s,fhz_s,Fs_s,tdur_s,phaseShift_s);
						
				int FFTlength=2048;
				double[][] frequencySpectrum_Rect=new double[2][FFTlength];
				double[][] frequencySpectrum_Hann=new double[2][FFTlength];
				double[] Magnitude_Lin_Rect=new double[FFTlength];
				double[] Magnitude_Lin_Hann=new double[FFTlength];
				double[] Magnitude_dB_Rect=new double[FFTlength];
				double[] Magnitude_dB_Hann=new double[FFTlength];
				double[] faxis=new double[FFTlength];
				int sigLen=2000;
				double[] rectangular=new double[sigLen];
				double[] winSigRect=new double[sigLen];
				double[] hanning=new double[sigLen];
				double[] winSigHann=new double[sigLen];
				
				rectangular=fourierTransform.rectWindow(sigLen);
				hanning=fourierTransform.hannWindow(sigLen);
				winSigRect=fourierTransform.applyWindow(output1s, rectangular, sigLen);
				winSigHann=fourierTransform.applyWindow(output1s, hanning, sigLen);
				frequencySpectrum_Rect=fourierTransform.doFFT(winSigRect, FFTlength);
				frequencySpectrum_Hann=fourierTransform.doFFT(winSigHann, FFTlength);
				boolean halfSpectrum=true;
				Magnitude_Lin_Rect=fourierTransform.SpecMagnitude(frequencySpectrum_Rect,halfSpectrum);
				Magnitude_Lin_Hann=fourierTransform.SpecMagnitude(frequencySpectrum_Hann,halfSpectrum);
				Magnitude_dB_Rect=fourierTransform.convert2dB(Magnitude_Lin_Rect);
				Magnitude_dB_Hann=fourierTransform.convert2dB(Magnitude_Lin_Hann);
				faxis=fourierTransform.makeFrequencyAxis(Fs_s,FFTlength,halfSpectrum);
				makePlot_s.plotData(faxis, Magnitude_dB_Rect,"Magnitude of Frequency Spectrum of Sinewave with FFT length=2048, Rectangular window", "Frequency (Hz)", "Magnitude", "Spectral Plots");
						
				makePlot_s.plotData(faxis, Magnitude_dB_Hann, "Magnitude of Frequency Spectrum of Sinewave with FFT length=2048, Hanning window", "Frequency (Hz)", "Magnitude", "Spectral Plots");
				} else if (plotNumber==6) { //Windowing a multisine signal
					signalGen sig_s=new signalGen();
					plotXYline makePlot_s=new plotXYline();
					FFT fourierTransform=new FFT();
					
					double Fs_m=11025;
					double[] fhzVect_m= {110.0, 220.0, 330.0,440.0,550.0,660.0};
					double tdur_m=1;
					double[] amplitudeVect_m= {1,0.5,0.333,0.25,0.2,0.167};
					double[] phaseShiftVect_m= {0.0,0.0,0.0,0.0,0.0,0.0};
					double[] output1m=sig_s.MultiSine(amplitudeVect_m,fhzVect_m,Fs_m,tdur_m,phaseShiftVect_m);
												
					int FFTlength=8192;
					double[][] frequencySpectrum_Rect=new double[2][FFTlength];
					double[][] frequencySpectrum_Hann=new double[2][FFTlength];
					double[] Magnitude_Lin_Rect=new double[FFTlength];
					double[] Magnitude_Lin_Hann=new double[FFTlength];
					double[] Magnitude_dB_Rect=new double[FFTlength];
					double[] Magnitude_dB_Hann=new double[FFTlength];
					double[] faxis=new double[FFTlength];
					int sigLen=4000;
					double[] rectangular=new double[sigLen];
					double[] winSigRect=new double[sigLen];
					double[] hanning=new double[sigLen];
					double[] winSigHann=new double[sigLen];
					
					rectangular=fourierTransform.rectWindow(sigLen);
					hanning=fourierTransform.hannWindow(sigLen);
					winSigRect=fourierTransform.applyWindow(output1m, rectangular, sigLen);
					winSigHann=fourierTransform.applyWindow(output1m, hanning, sigLen);
					frequencySpectrum_Rect=fourierTransform.doFFT(winSigRect, FFTlength);
					frequencySpectrum_Hann=fourierTransform.doFFT(winSigHann, FFTlength);
					boolean halfSpectrum=true;
					Magnitude_Lin_Rect=fourierTransform.SpecMagnitude(frequencySpectrum_Rect,halfSpectrum);
					Magnitude_Lin_Hann=fourierTransform.SpecMagnitude(frequencySpectrum_Hann,halfSpectrum);
					Magnitude_dB_Rect=fourierTransform.convert2dB(Magnitude_Lin_Rect);
					Magnitude_dB_Hann=fourierTransform.convert2dB(Magnitude_Lin_Hann);
					faxis=fourierTransform.makeFrequencyAxis(Fs_m,FFTlength,halfSpectrum);
					
					makePlot_s.plotData(faxis, Magnitude_dB_Rect,"Magnitude of Frequency Spectrum of MultiSine with FFT length=8192, Rectangular window", "Frequency (Hz)", "Magnitude", "Spectral Plots");	
					makePlot_s.plotData(faxis, Magnitude_dB_Hann, "Magnitude of Frequency Spectrum of MultiSine with FFT length=8192, Hanning window", "Frequency (Hz)", "Magnitude", "Spectral Plots");
					System.out.println("Using a rectangular window the spectral sidelobes/noise floor is much higher");
					System.out.println("Using a Hanning window the spectral sidelobes/noise floor are lower");
				} else if (plotNumber==7) { //White Noise
					signalGen sig_s=new signalGen();
					plotXYline makePlot_s=new plotXYline();
					FFT fourierTransform=new FFT();
				    
					int sigLen=11025;
					double Fs_w=11025;
					double[] wn=sig_s.whiteNoise(sigLen);
					int FFTlength=4096;
					
					double[][] frequencySpectrum=new double[2][FFTlength];
					double[] Magnitude_Lin=new double[FFTlength];
					double[] Magnitude_dB=new double[FFTlength];
					double[] faxis=new double[FFTlength];
					
					frequencySpectrum=fourierTransform.doFFT(wn, FFTlength);
					boolean halfSpectrum=true;
					Magnitude_Lin=fourierTransform.SpecMagnitude(frequencySpectrum,halfSpectrum);
					Magnitude_dB=fourierTransform.convert2dB(Magnitude_Lin);
					faxis	=fourierTransform.makeFrequencyAxis(Fs_w,FFTlength,halfSpectrum);
					makePlot_s.plotData(faxis, Magnitude_dB, "Magnitude of Frequency Spectrum of White Noise FFT length=4096", "Frequency (Hz)", "Magnitude", "Spectral Plots");
	}
	}	
}
